grinpy/functions/structural_properties.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+
+#    Copyright (C) 2017 by
+#    David Amos <somacdivad@gmail.com>
+#    Randy Davila <davilar@uhd.edu>
+#    BSD license.
+#
+# Authors: David Amos <somacdivad@gmail.com>
+#          Randy Davila <davilar@uhd.edu>
+"""Functions for computing various structural properites."""
+
+import networkx as nx
+import itertools
+
+def is_triangle_free(G):
+    """Returns a boolean value according to G containing an induced triangle.
+
+    Parameters
+    ----------
+    G : graph
+        A NetworkX graph.
+
+    Returns
+    -------
+    is_triangle_free : boolean
+                       True if G containes no induced triangle. False if G contains at least one induced triangle.
+
+    Examples
+    --------
+    >>> G = nx.Graph([(0,1),(0,2),(0,3),(1,2),(1,3),(2,3)]) # The complete graph K_4
+    >>> nx.is_triangle_free(G)
+    False
+    """
+    # define a triangle graph, also known as the complete graph K_3
+    triangle = nx.Graph([(0,1),(1,2),(2,0)])
+    
+    # assume first that G is triangle-free
+    triangle_free = True
```

Comment:
This line is extraneous. The loop below will _only_ complete if the graph is triangle free, so you can delete this line and then change the final `return` statement to `return True`.